internal/driver/common/driver.go

```diff
@@ -0,0 +1,19 @@
+package common
+
+import (
+	"sync"
+
+	"fyne.io/fyne/v2"
+)
+
+var (
+	canvasMutex sync.RWMutex
+	canvases    = make(map[fyne.CanvasObject]fyne.Canvas)
+)
+
+// CanvasForObject returns the canvas for the specified object.
+func CanvasForObject(obj fyne.CanvasObject) fyne.Canvas {
+	canvasMutex.RLock()
+	defer canvasMutex.RUnlock()
+	return canvases[obj]
```

Comment:
I am no master when it comes to mutexes, but are these locks really going to work? Wouldn't we have to assign `canvases[obj]` to a new variable in-between?

Reply:
Hmm, not sure what you mean 🤔. The mutex is there, to ensure we read atomically the canvases map (ensuring we are not writing and reading it at the same time). I think, assigning it to a variable does not make any difference (when you are returning the value, you are already creating a new variable to assign it) 🤔

Reply:
I don't know. I thought defers ran before returning the value, but I might be very wrong about that.

Reply:
Oh I see, `defers` ran after returning the value 😄:
https://play.golang.org/p/AhQBXxb1HAH

Reply:
Ah, sorry then 😅

Reply:
No worries :)